### PR TITLE
feat: fix warnings, repl features, request_options fixes, and atnotification refactor

### PR DESCRIPTION
**Pull request by JeremyTubongbanua**
**- What I did**
- fixed "redefinition" warnings from connection_hooks/connection
- fixed other misc warnings
- fixed warnings with unused parameters in static argument validation functions throughout the SDK
- New debug_build.sh for bulding the C SDK with `-DCMAKE_BUILD_TYPE=Debug`

REPL changes:
- edited run.sh
- new debug_build.sh for repl
- edited CMakeLists.txt to remove an unused include()
- added /put to repl
- added /delete to repl

atchops/rsa_key:
- New `atchops_rsa_key_is_public_key_populated` function
- New `atchops_rsa_key_is_private_key_populated` function
- Made `rsa_key_is_*_initialized` boolean checkers receive a const pointer instead of a non-const pointer

atclient/atnotitication
- new function `int atclient_atnotification_from_json_str(atclient_atnotification *notification, const char *json_str);`
- new function `int atclient_atnotification_from_cjson_node(atclient_atnotification *notification, const cJSON *root);`

request_options
- made the `set_initialized` functions to static within the `.c`. we don't want to expose this because it is internal memory management

**- Description for the changelog**
fix: warnings

### CODE REVIEW

**Comment by JeremyTubongbanua** [comment]
Converting to draft as I fix NoPorts C Daemon

File: packages/atclient/include/atclient/connection_hooks.h [hunk 1]
```diff
@@ -44,18 +46,18 @@ typedef struct atclient_connection_hooks {
...
+int atclient_connection_hooks_set(struct atclient_connection *ctx, const atclient_connection_hook_type type, void *hook);
 
-bool atclient_connection_hooks_is_pre_read_initialized(const atclient_connection *ctx);
-bool atclient_connection_hooks_is_post_read_initialized(const atclient_connection *ctx);
-bool atclient_connection_hooks_is_pre_write_initialized(const atclient_connection *ctx);
-bool atclient_connection_hooks_is_post_write_initialized(const atclient_connection *ctx);
+bool atclient_connection_hooks_is_pre_read_initialized(const struct atclient_connection *ctx);
+bool atclient_connection_hooks_is_post_read_initialized(const struct atclient_connection *ctx);
+bool atclient_connection_hooks_is_pre_write_initialized(const struct atclient_connection *ctx);
+bool atclient_connection_hooks_is_post_write_initialized(const struct atclient_connection *ctx);
```
Comment (xavierchanth) [inline packages/atclient/include/atclient/connection_hooks.h:60]: .

File: packages/atclient/include/atclient/connection_hooks.h [hunk 2]
```diff
@@ -17,6 +17,8 @@
 #define ATCLIENT_CONNECTION_HOOKS_PRE_WRITE_INITIALIZED (VALUE_INITIALIZED << 2)
 #define ATCLIENT_CONNECTION_HOOKS_POST_WRITE_INITIALIZED (VALUE_INITIALIZED << 3)
 
+struct atclient_connection;
+
```
Comment (xavierchanth) [inline packages/atclient/include/atclient/connection_hooks.h:21]: .

**Comment by JeremyTubongbanua** [comment]
I have updated the PR description @realvarx @XavierChanth.. this is ready for review.

**Comment by JeremyTubongbanua** [comment]
NoPorts' C SSHNPD commit hash is set to `24f916e7379011f186b5f357e4bfa3b0ff2d4f6a` which is the latest commit as of this message.

**Comment by JeremyTubongbanua** [comment]
> NoPorts' C SSHNPD commit hash is set to `24f916e7379011f186b5f357e4bfa3b0ff2d4f6a` which is the latest commit as of this message.

Sorry I meant to say `dc9fc44e789b91887a5be13c433d6039b72aaea6`

**Review by xavierchanth — APPROVED** [review]

**Comment by xavierchanth** [comment]
Nice work

**Pull request merged**